bin/wkhtmltopdf

```diff
@@ -50,4 +51,24 @@ unless File.exist? binary
         "(missing binary: #{binary})."
 end
 
-exec *$*.unshift(binary)
+raise 'chromium-browser binary not found.' unless system('which chromium-browser')
+
+uuid = SecureRandom.uuid
+# ENV['PWD'] is path to kings-landing
+temp_path = "#{ENV['PWD']}/tmp"
+
+html_path = "#{temp_path}/html_#{uuid}.html"
+File.open(html_path, 'wb') do |f|
+  f.write "\uFEFF"
+  f.write STDIN.read
+end
```

Comment:
close open files when done

Reply:
I don't understand. File is closed when you reach the end statement on line 64 or am I missing something?

Reply:
unless the process terminates, open file descriptors are not closed by unix kernel AFAIK

Reply:
Only if we leave it to GC without close statement, something like this:
```
my_file = File.open(...)
my_file.write ...
```
Then yes, you need to call my_file.close explicitly.
When you provide a block to File.open, close method is called when the block is terminated.

Reply:
@hj-glooko ruby docs explain it pretty nice:

> With no associated block, File.open is a synonym for ::new. If the optional code block is given, it will be passed the opened file as an argument and the File object will automatically be closed when the block terminates.